website/api-gateway/src/json-rpc/handler.ts

```diff
@@ -1,88 +1,36 @@
-import { Observable } from 'rxjs';
-import { getResponse } from 'src/utils';
-import { MethodNotFoundError } from './errors';
-import { IRpcRequest, IRpcResponse } from './interface';
+import { IRpcRequest, IRpcResponse } from '@gear-js/interfaces';
 import errors from '@gear-js/jsonrpc-errors';
 
-@Injectable()
-export class RpcMessageHandler {
-  isArray = (message: any) => Array.isArray(message);
-  isObject = (message: any) => {
-    return typeof message === 'object' && message != null;
-  };
+import { getResponse } from '../utils';
+import { kafkaProducerTopics } from '../common/kafka-producer-topics';
+import { kafkaEventHandler } from '../helpers/kafka-event.handler';
+import { logger } from '../helpers/logger';
 
-  methods: {};
-
-  checkMethod(procedure: IRpcRequest): (params: any) => Observable<any> {
-    const method = procedure.method;
-    if (!method) {
-      throw new MethodNotFoundError();
-    } else {
-      return this.getMethod(method);
-    }
-  }
-
-  getMethod(methodName: string): (params: any) => Observable<any> {
-    const method = methodName.split('.');
-    let result = Object.getOwnPropertyDescriptor(this.methods, method[0]);
-    if (result && method.length > 1) {
-      method.slice(1).forEach((element) => {
-        result = Object.getOwnPropertyDescriptor(result.value, element);
-        if (!result) {
-          return undefined;
-        }
-      });
-    }
-    return result ? result.value : result;
-  }
-
-  async requestMessage(message: IRpcRequest | IRpcRequest[]): Promise<IRpcResponse | IRpcResponse[]> {
-    if (message instanceof Array) {
-      const results = [];
-      const promises = message.map(async (procedure) => {
-        const result = await this.executeProcedure(procedure);
-        results.push(result);
-      });
-      await Promise.all(promises);
-      return results;
-    } else {
-      return this.executeProcedure(message);
-    }
+async function requestMessageHandler(message: IRpcRequest | IRpcRequest[]): Promise<IRpcResponse | IRpcResponse[]> {
+  if (Array.isArray(message)) {
+    const promises = message.map((rpcMessage) => {
+      return executeProcedure(rpcMessage);
+    });
+    const result = await Promise.all(promises);
+    return result;
+  } else {
+    return executeProcedure(message);
   }
+}
 
-  async executeProcedure(procedure: IRpcRequest): Promise<IRpcResponse> {
-    let response = null;
-    try {
-      const method = this.checkMethod(procedure);
-      if (!method) {
-        throw new MethodNotFoundError();
-      }
-      response = this.executeMethod(method, procedure);
-    } catch (error) {
-      response = getResponse(procedure, error.toJson());
-    }
-    return response;
+async function executeProcedure(procedure: IRpcRequest): Promise<IRpcResponse> {
+  if (!isExistMethod(procedure.method)) {
+    logger.error(JSON.stringify(errors.MethodNotFound));
+    return getResponse(procedure, 'MethodNotFound');
```

Comment:
```suggestion
    return getResponse(procedure, errors.MethodNotFound.name);
```
what do you think?

Reply:
I agree with you. It will be better to use this version `errors.MethodNotFound.name`.